### ISSUE TITLE
crash on powerful windows computer

### ISSUE THREAD

**Issue by ajdamico** [boennecd/parglm#3]
pretty simple code to reproduce:

	library(parglm)
	this_df <- data.frame( a = sample( 1:1000000 , 20 ) / 100 , b = 1 )
	parglm( a ~ b , data = this_df , nthreads = parallel::detectCores() ) # sometimes you need to run this line five or six times
	
	
	
this only occurs on a powerful computer, not on my local machine

	Microsoft Windows [Version 6.3.9600]
	(c) 2013 Microsoft Corporation. All rights reserved.

	C:\Users\AnthonyD>"c:\Program Files\r\R-3.5.2\bin\x64\Rterm.exe"

	R version 3.5.2 (2018-12-20) -- "Eggshell Igloo"
	Copyright (C) 2018 The R Foundation for Statistical Computing
	Platform: x86_64-w64-mingw32/x64 (64-bit)

	R is free software and comes with ABSOLUTELY NO WARRANTY.
	You are welcome to redistribute it under certain conditions.
	Type 'license()' or 'licence()' for distribution details.

	  Natural language support but running in an English locale

	R is a collaborative project with many contributors.
	Type 'contributors()' for more information and
	'citation()' on how to cite R or R packages in publications.

	Type 'demo()' for some demos, 'help()' for on-line help, or
	'help.start()' for an HTML browser interface to help.
	Type 'q()' to quit R.

	>
	> sessionInfo()
	R version 3.5.2 (2018-12-20)
	Platform: x86_64-w64-mingw32/x64 (64-bit)
	Running under: Windows Server 2012 R2 x64 (build 9600)

	Matrix products: default

	locale:
	[1] LC_COLLATE=English_United States.1252
	[2] LC_CTYPE=English_United States.1252
	[3] LC_MONETARY=English_United States.1252
	[4] LC_NUMERIC=C
	[5] LC_TIME=English_United States.1252

	attached base packages:
	[1] stats     graphics  grDevices utils     datasets  methods   base

	loaded via a namespace (and not attached):
	[1] compiler_3.5.2
	> parallel::detectCores()
	[1] 64
	> library(parglm)
	> this_df <- data.frame( a = sample( 1:1000000 , 20 ) / 100 , b = 1 )
	> parglm( a ~ b , data = this_df , nthreads = parallel::detectCores() )

	C:\Users\AnthonyD>

**Comment by boennecd** [comment]
This also yields an error on Ubuntu 18.04:

```r
library(parglm)
for(i in 1:100000){
  this_df <- data.frame( a = sample( 1:1000000 , 20 ) / 100 , b = 1 )
  parglm( a ~ b , data = this_df , nthreads = 64)
}
```

I never tested the code where you have less rows than threads. This may be the source of the error since this does not crash 

```r
library(parglm)
for(i in 1:100000){
  this_df <- data.frame( a = sample( 1:1000000 , 100 ) / 100 , b = 1 )
  parglm( a ~ b , data = this_df , nthreads = 64)
}
```

**Comment by ajdamico** [comment]
for me it's a crash, not an error..

**Comment by boennecd** [comment]
I mean that it is an error that I did not test the code in situations where there are less rows than threads. It is something that I will fix. I plan to 

1. return a warning if the user have sets less than x rows per thread with the comment that there will likely not be speed up in this situation.  
2. Set the number of threads such that there is at least x rows per thread. 

and handle a situation with less than x rows as a special case.

**Comment by boennecd** [comment]
Fixed in bd22d86c400d3b06b6f877ea2ad672835af7748f. Now this code does not crash

```r
this_df <- data.frame( a = sample( 1:1000000 , 20 ) / 100 , b = 1 )
fit <- parglm( a ~ b , data = this_df , nthreads = 64)
#R Warning message:
#R In method(x = c(1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,  :
#R   Too few observation compared to the number of threads. 2 thread(s) will be used instead of 64.

for(i in 1:10000)
  suppressWarnings(parglm( a ~ b , data = this_df , nthreads = 64))
```

**Comment by ajdamico** [comment]
no more crash for me, thank you!
